lib/Index/IndexRecord.cpp

```diff
@@ -393,6 +396,195 @@ emitDataForSwiftSerializedModule(ModuleDecl *module,
                                  const PathRemapper &pathRemapper,
                                  SourceFile *initialFile);
 
+static void
+appendSymbolicInterfaceToIndexStorePath(SmallVectorImpl<char> &resultingPath) {
+  llvm::sys::path::append(resultingPath, "interfaces");
+}
+
+static bool initSymbolicInterfaceStorePath(StringRef storePath,
```

Comment:
Could we use `llvm::Error` instead?
```
static llvm::Error initSymbolicInterfaceStorePath(StringRef storePath) {
  SmallString<128> subPath = storePath;
  appendSymbolicInterfaceToIndexStorePath(subPath);
  if (std::error_code ec = llvm::sys::fs::create_directories(subPath))
    return llvm::errorCodeToError(ec);
  return llvm::Error::success();
}
```

And maybe skip the `llvm::...` if we already declare those in the swift namespace (I didn't check).